### PR TITLE
Add basic tests and CI pipeline

### PR DESCRIPTION
**Pull request by twolodzko**
- I added a CI pipeline like in other repos.
- I added a basic test based on the quick start example from https://optuna.org/.

### CODE REVIEW

File: tests/test_e2e.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import neptune_optuna.impl as npt_utils
+
+def test_e2e():
+
+    n_trials = 5
+    base_namespace = "optuna"
+
+    run = neptune.init_run()
+    # handler = run[base_namespace]
```
Comment (kshitij12345) [inline tests/test_e2e.py:18]: I think we want to add another test with `handler` instead of `run`?

Comment (twolodzko) [reply tests/test_e2e.py:18]: Yes, but that is not supported with the current codebase and here I am only adding tests to the existing code.

**Review by kshitij12345 — APPROVED** [review]
LGTM! Thanks @twolodzko

File: tests/test_e2e.py (hunk 1 above)
Comment (kshitij12345) [reply tests/test_e2e.py:18]: Got it!

**Pull request merged**